### DISCUSSION TITLE
SetValue does not work properly after reseting the form using reset() function

### DISCUSSION THREAD

**Discussion by ashishkkhatri** [react-hook-form/react-hook-form#5898] [Q&A] [answered]
There are 2 date fields(transactionDate and documentDate) on change of transaction date other one should be updated automatically. Which i am able to do using setValue function from useForm.

I have a used case where on change of a dropdown i have to reset the from which i am able to do by using reset function. But after the form is reset, onChange of transactionDate i update the value for documentDate using setValue and this doesn't work.

[here](https://codesandbox.io/s/amazing-framework-pi4ot?file=/src/App.js) is the sample code.

This issue I am facing in react hook form v7.10.1

**Comment by bluebill1049** [comment, marked as answer]
why not just pass value to `reset` and let it reset the value?

**Reply by ashishkkhatri** [reply]
First of all thanks for quick response.
reset generally would be used to set the values for whole form right. if i have to update value for one field i think using setValue is better option rather than setting the value for whole form again.

**Reply by bluebill1049** [reply]
i guess if you are already reset the whole form, it doesn't matter anymore. 

```
reset({
   ...getValues(),
   // your fireld
})
```

see you doing one step less  as well.

**Reply by ashishkkhatri** [reply]
Ok.. but why are we restricting the developer to not to use setValue once he/she has uses reset.

**Reply by bluebill1049** [reply]
not restricting... because it's a procedure reset needs to happen first then setValue...

**Reply by ashishkkhatri** [reply]
thank you so much @bluebill1049

**Reply by bluebill1049** [reply]
You are welcome.

**Reply by ashishkkhatri** [reply]
Hi @bluebill1049,

one more scenario regarding reset i am facing. Since i am using reset to set the initial value i will have to use reset to update even one field as i cannot use setValue after reset. By this approach every time i do reset all the watchers in the form will detect a change and respective useEffect callbacks whose dependencies are mentions as these watchers will be triggered. Here i dont want all the watchers to detect the change instead i want only those watchers to detect the change whose values are actually changed.

Need your help here
